compiler/rustc_infer/src/infer/lexical_region_resolve/mod.rs

```diff
@@ -882,12 +880,19 @@ impl<'tcx> LexicalRegionResolutions<'tcx> {
         &mut self.values[rid]
     }
 
-    pub fn resolve_var(&self, rid: RegionVid) -> ty::Region<'tcx> {
-        let result = match self.values[rid] {
-            VarValue::Value(r) => r,
-            VarValue::ErrorValue => self.error_region,
+    pub(crate) fn resolve_region(
+        &self,
+        tcx: TyCtxt<'tcx>,
+        r: ty::Region<'tcx>,
+    ) -> ty::Region<'tcx> {
+        let result = match *r {
```

Comment:
I'm realizing that right now, this match is redundant, but in future work, I've made it so that it's necessary. I could revert this, but the point was to make the followup changes smaller.

Reply:
I've changed the only fully redundant bit to be redundant.